content/guides/core-concepts/writing-and-organizing-tests.md

```diff
@@ -545,6 +545,60 @@ it.skip('returns "fizz" when number is multiple of 3', () => {
 })
 ```
 
+### Test Isolation
+
+<Alert type="success">
+
+<Icon name="check-circle" color="green"></Icon> **Best Practice:** Clean up
+state **before** tests run.
+
+</Alert>
+
+Test isolation is the practice of resetting the state _before_ each test.
+
+Cleaning up state ensures the operations of one test does not affect another
+test downstream. When writing tests, the goal for the test is reliably pass when
+ran standalone or in a randomized order. If you have set up your tests in a way
+that requires the state of a previous test, this could potentially break your
+tests.
+
+Cypress supports two modes of test isolation, `default` and `strict`.
+
+#### Default Mode
+
+When in `default` mode, Cypress handles resetting the state for:
+
+- [aliases](/api/commands/as)
+- [cookies](/api/commands/clearcookies)
+- [clock](/api/commands/clock)
+- [intercepts](/api/commands/intercepts)
+- [localStorage](/api/commands/clearlocalstorage)
+- [routes](/api/commands/route)
+- [sessions](/api/commands/session)
+- [spies](/api/commands/spy)
+- [stubs](/api/commands/stub)
+- [viewport](/api/commands/viewport)
+
+#### Strict Mode
+
+<Alert type="warning">
+
+<strong class="alert-header"><Icon name="exclamation-triangle"></Icon>
+Experimental</strong>
+
+Strict mode is currently experimental and can be enabled by setting
+the [`experimentalSessionAndOrigin`](/guides/references/experiments) flag
+to `true` in the Cypress config. This is the default test isolation behavior
+when using the `experimentalSessionAndOrigin` experiment.
+
+</Alert>
+
+When in `strict` mode, Cypress handles resetting the state for everything
+outlined above for `default` mode, in addition to clearing the page by visiting
+`about:blank` before each test. This clears the current application state by
+forcing the tester to re-build the application's state for the test to ensure
+the test can reliably pass when ran standalone or in a randomized order.
```

Comment:
> This clears the current application state by forcing the tester to re-build the application's state for the test

This is not quite true.

Setting the page to `about:blank` clears non-persistent browser state only. It does not affect localStorage or cookies (which is fine because Cypress already does that) but it also has no effect on any backend state. So if they are running their frontend against a stateful server of some kind, e.g. a REST API connected to a database, this still contains application state and still needs to be reset before every test if complete test isolation is to be achieved.

Not sure how much of this we need to include here, but we definitely shouldn't imply that blanking the page will clear 100% of application state.

Reply:
gosh you are right...this is how we've been describing it to try to clarify and categorize this why we need to visit about:blank but theres def more than just the rendered dom that matters to fully clear the application. thanks for calling this out!